### PR TITLE
fix: Content-Type and Content-Disposition's filename were incorrect for Content-Encoding: gzip

### PR DESCRIPTION
**Pull request by jpmckinney**

### CODE REVIEW

File: exporter/views.py [hunk 1]
```diff
@@ -20,22 +20,28 @@ def download_export(request):
...
-    if suffix in ("jsonl.gz", "csv.tar.gz"):
-        headers = {"Content-Encoding": "gzip"}
+    if suffix == "jsonl.gz":
+        # https://stackoverflow.com/a/67807011/244258
+        headers = {"Content-Encoding": "gzip", "Content-Type": "application/jsonlines+json"}
+        extension = "jsonl"
+    if suffix == "csv.tar.gz":
+        # https://stackoverflow.com/a/57954148/244258
+        headers = {"Content-Encoding": "gzip", "Content-Type": "application/x-tar"}
+        extension = "csv.tar"
```
Comment (yolile) [inline exporter/views.py:30]:
```suggestion
        extension = "csv.tar.gz"
```

Otherwise the file is not found

**Review by yolile — CHANGES_REQUESTED** [review]

File: exporter/views.py (hunk 1 above)
Comment (jpmckinney) [reply exporter/views.py:30]: Ah, at the line `Export(job_id, basename=f"{stem}.{extension}")` it needs to be `Export(job_id, basename=f"{stem}.{suffix}")

The `filename` argument later needs be the uncompressed `extension`.

**Comment by jpmckinney** [comment]
I started adding some tests. Until Django 4.1 (https://github.com/django/django/pull/11414), FileResponse overrides Content-Type (and it has to be set via `content_type="xxx"` not `headers={"Content-Type": "xxx"}`). In any case, we shouldn't set Content-Encoding: gzip unless the client sends Accept-Encoding: gzip. So, I will change this PR to instead circumvent GzipMiddleware when serving these files.

**Review by yolile — APPROVED** [review]

**Comment by yolile** [comment]
Deployed

**Pull request merged**